pkg/deployments/tasks/20230403-timelock-authorizer/test/task.fork.ts

```diff
@@ -0,0 +1,153 @@
+      const authorizerTask = new Task('20210418-authorizer', TaskMode.READ_ONLY, getForkedNetwork(hre));
+      oldAuthorizer = await authorizerTask.instanceAt('Authorizer', await migrator.oldAuthorizer());
+
+      const multisig = await impersonate(input.Root, fp(100));
+      const setAuthorizerActionId = await actionId(vault, 'setAuthorizer');
+      await oldAuthorizer.connect(multisig).grantRolesToMany([setAuthorizerActionId], [migrator.address]);
+    });
+
+    it('migrates all roles properly', async () => {
+      for (const roleData of await getOnChainRoles(
+        input.Roles,
+        input.TRANSITION_START_BLOCK,
+        input.TRANSITION_END_BLOCK
+      )) {
+        expect(await newAuthorizer.hasPermission(roleData.role, roleData.grantee, roleData.target)).to.be.true;
+      }
+    });
+
+    it('sets up granters properly', async () => {
+      for (const granterData of input.Granters) {
+        expect(await newAuthorizer.isGranter(granterData.role, granterData.grantee, granterData.target)).to.be.true;
+      }
+    });
+
+    it('sets up revokers properly', async () => {
+      for (const revokerData of input.Revokers) {
+        expect(await newAuthorizer.isRevoker(revokerData.role, revokerData.grantee, revokerData.target)).to.be.true;
+      }
+    });
+
+    it('sets up delays properly', async () => {
+      await advanceTime(5 * DAY);
+      await migrator.executeDelays();
+
+      for (const delayData of input.ExecuteDelays) {
+        expect(await newAuthorizer.getActionIdDelay(delayData.actionId)).to.be.eq(delayData.newDelay);
+      }
+
+      for (const delayData of input.GrantDelays) {
+        expect(await newAuthorizer.getActionIdGrantDelay(delayData.actionId)).to.be.eq(delayData.newDelay);
+      }
+    });
+
+    it('does not set the new authorizer immediately', async () => {
+      expect(await newAuthorizer.isRoot(migrator.address)).to.be.true;
+      expect(await vault.getAuthorizer()).to.be.equal(oldAuthorizer.address);
+    });
+
+    it('finalizes the migration once new root address claims root status', async () => {
+      await expect(migrator.finalizeMigration()).to.be.revertedWith('ROOT_NOT_CLAIMED_YET');
+
+      await newAuthorizer.connect(root).claimRoot();
+
+      await migrator.finalizeMigration();
+      expect(await vault.getAuthorizer()).to.be.equal(newAuthorizer.address);
+      expect(await newAuthorizer.isRoot(root.address)).to.be.true;
+      expect(await newAuthorizer.isRoot(migrator.address)).to.be.false;
+    });
+
+    it('allows minting after the migration', async () => {
+      const balancerMinterTask = new Task('20220325-gauge-controller', TaskMode.READ_ONLY, getForkedNetwork(hre));
+      const balancerMinter = await balancerMinterTask.deployedInstance('BalancerMinter');
+
+      const balancerTokenAdminTask = new Task(
+        '20220325-balancer-token-admin',
+        TaskMode.READ_ONLY,
+        getForkedNetwork(hre)
```

Comment:
Does this work on goerli? I didn't remember we had set it up.

At any rate, it'll definitely fail on the L2s (and also is irrelevant). Can we extract it out of the generic tests?